Implement changes for SDLC - CI pipeline and PR template

AlexioDanje opened the pull request:
- Add Vitest coverage configuration with initial thresholds
- Setup Code Climate test reporter integration
- Configure coverage directory handling in CI workflow
- Ensure proper test coverage reporting both locally and in CI

Testing:
- Verified local coverage generation
- Confirmed CI pipeline uploads reports to Code Climate

Commit:
ci: Configure CI pipeline to handle coverage reporting

- Update vitest.config.js with coverage settings and JSX transformation
- Configure CI workflow to create coverage directories
- Set initial coverage thresholds to 0 for gradual improvement
- Add Code Climate test reporter integration

The changes ensure proper test coverage reporting both locally and in CI,
with Code Climate integration for tracking coverage metrics.

timler commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :no_entry: **Snyk checks have failed. 1 issues have been found so far.**

| Icon     | Severity | Issues  |
| -------- | -------- | ------- |
| ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) | Critical | 0 |
| ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) | High | 0 |
| ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) | **Medium** | **1** |
| ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) | Low | 0 |

:no_entry: **security/snyk** check is complete. 1 issues have been found. [(View Details)](https://app.snyk.io/org/decode/pr-checks/2c89372e-aea4-45e5-b88e-2c6ad579771d?refs=ac0e8a640dce1fce06e3a9063b5bf9b6b91454f1&source=prChecksComment)

timler commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,78 @@
+      - '**.md'
+      - 'docs/**'
+      - '.gitignore'
+      - 'LICENSE'
```
I will add this to my workflow so they match

timler commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,78 @@
+  NODE_VERSION: '16'
+
+jobs:
+  continuous-integration:
```
I think we should align our workflows - please change this to "test" to match the job on npm "test:coverage", and then change the name of the file to continuous-integration.yml

AlexioDanje replied:
Thank you @timler I have updated my PR

timler commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,78 @@
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
I forgot to add this for my pipeline, so will do that!

timler reviewed (commented):
Looks good - minor comments.

Commit:
ci: align workflow naming conventions and add PR template

- Rename workflow file to continuous-integration.yml
- Update job name to test to match npm script (test:coverage)
- Add pull request template for standardization

Commit:
ci: align workflow naming conventions and add PR template

- Update the folder structrure for the pull request template

timler commented on .github/workflows/continuous-integration.yml:
```diff
@@ -0,0 +1,78 @@
+      - '**.md'
+      - 'docs/**'
+      - '.gitignore'
+      - 'LICENSE'
```
This is what I added - AFAIK you don't need separate definition for pull_request because it's covered under push

```
on:
  push:
    paths-ignore:
      - '**.md'
      - '**.txt'
      - '**.json'
      - '**.yml'
      - '**.properties'
      - '**.xml'
      - '**.sql'
      - '**.csv'
      - '**.zip'
      - 'docs/**'
      - '.gitignore'
```

AlexioDanje replied:
It makes sense, let me resolve that

Commit:
ci: refine CI pipeline configuration

- Update paths-ignore to ensure critical files trigger builds
- Maintain focus on push events, the pull request trigger is redundant

timler committed:
Moved the PR template

Moved the PR template from the folder into the main .github folder so
the template is applied by default and doesn't need a query parameter
